### PR TITLE
feat: [CDS-69341]: add find user email api for github in go-scm

### PR DESCRIPTION
**Pull request by shalini-agr**

### CODE REVIEW

File: scm/driver/github/user.go [hunk 1]
```diff
@@ -31,8 +31,9 @@ func (s *userService) FindLogin(ctx context.Context, login string) (*scm.User, *
 }
 
 func (s *userService) FindEmail(ctx context.Context) (string, *scm.Response, error) {
-	user, res, err := s.Find(ctx)
-	return user.Email, res, err
```
Comment (bradrydzewski) [inline scm/driver/github/user.go:35]: you can re-use the `ListEmail` function here:

```
func (s *userService) FindEmail(ctx context.Context) (string, *scm.Response, error) {
	out, res, err := s.ListEmails(ctx, scm.ListOptions{})
	return returnPrimaryEmail(out), res, err
}
```

File: scm/driver/github/user_test.go [hunk 2]
```diff
@@ -87,14 +87,14 @@ func TestUserEmailFind(t *testing.T) {
...
-		Get("/user").
+		Get("/user/emails").
 		Reply(200).
 		Type("application/json").
 		SetHeader("X-GitHub-Request-Id", "DD0E:6011:12F21A8:1926790:5A2064E2").
 		SetHeader("X-RateLimit-Limit", "60").
 		SetHeader("X-RateLimit-Remaining", "59").
 		SetHeader("X-RateLimit-Reset", "1512076018").
-		File("testdata/user.json")
+		File("testdata/userEmail.json")
```
Comment (bradrydzewski) [inline scm/driver/github/user_test.go:97]: files in the testdata folder use snake case for file names as opposed to camel case (e.g. `user_email.json` as opposed to `userEmail.json`) however, you can probably re-use the

Comment (bradrydzewski) [reply scm/driver/github/user_test.go:97]: although you may be able to re-use the existing `testdata/emails.json` file in that folder for your test

File: scm/driver/github/user.go (hunk 1 above)
Comment (shalini-agr) [reply scm/driver/github/user.go:35]: done

Comment (shalini-agr) [reply scm/driver/github/user.go:35]: done

File: scm/driver/github/user_test.go (hunk 2 above)
Comment (shalini-agr) [reply scm/driver/github/user_test.go]: done

**Review by mohitg0795 — APPROVED** [review]

**Pull request merged**